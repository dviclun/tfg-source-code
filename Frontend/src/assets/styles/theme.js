import { createTheme } from "@mui/material";
import { orange } from '@mui/material/colors';
import Poppins from '../fonts/Poppins-Regular.ttf'

export const theme = createTheme({
    status: {
        danger: orange[500],
      },
    palette: {
        slate100: '#f1f5f9',
        slate300:'#cbd5e1',
        green300: '#86efac',
        green400: '#4ade80',
        green500: '#22c55e',
        green600: '#16a34a',
        green700: '#047857', 
        green900: '#14532d',
        red400: '#f87171',
        red500: '#ef4444',
        zinc200: '#e4e4e7',
        zinc300: '#d4d4d8',
        zinc400: '#a1a1aa',
        rose400: '#fb7185',
        rose500: '#f43f5e',
        purple400: '#c084fc',
        purple500: '#a855f7',
        blue300: '#93c5fd',
        blue400: '#60a5fa',
        gray100: '#f3f4f6',
        neutral100: '#f5f5f5',
        neutral200: '#e5e5e5',
        emerald100: '#d1fae5'
    },
});