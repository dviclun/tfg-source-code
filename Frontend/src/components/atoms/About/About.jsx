import React from 'react'
import { Header } from '../../molecules/Header/Header'
import { Grid, Typography, useMediaQuery } from '@mui/material'
import LogoTrass from '../../../assets/trassierra.jpg';
import { useTheme } from '@emotion/react';
import { useTranslation } from 'react-i18next';


export const About = () => {
    
    const theme = useTheme();

    const isSmall = useMediaQuery(theme.breakpoints.down('sm'));

    const [t] = useTranslation("global");

  return (
    <>
    <Header/>
    <Grid container justifyContent={'center'} sx={{ marginTop: '80px', padding: '2em' }}>
        <Grid item xs={12}>
            <Typography variant={isSmall ? 'h5': 'h3'} sx={{fontWeight:'bold', fontFamily: 'Poppins', paddingBottom: '.5em', textAlign: 'center'}} className="poppins-medium" >NEW CONCEPT <span className="text-green-600 poppins-bold-italic">GYM</span></Typography> 
        </Grid>
        <Grid item xs={12}>
            <Typography variant='h6' sx={{textAlign:'center', fontStyle: 'italic',fontFamily: 'Poppins'}}>Daniel Vicent Luna</Typography>
        </Grid>
        <Grid item xs={12}>
            <Typography sx={{textAlign: 'center', fontFamily: 'Poppins', marginTop: '3em', fontWeight: 'bold'}}>{t("about.text")}</Typography>
        </Grid>
        <Grid item xs={12} display={'flex'} justifyContent={'center'} sx={{marginTop: '3em'}}>
            <img src={LogoTrass} alt='IES Trassierra Logo'></img>
        </Grid>
    </Grid>
    </>
  )
}
