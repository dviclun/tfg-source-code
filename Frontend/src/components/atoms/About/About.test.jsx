import { render } from "@testing-library/react"
import { About } from "./About"
import { ThemeProvider } from "@emotion/react"
import { theme } from "../../../assets/styles/theme"
import { BrowserRouter } from "react-router-dom"
import { AuthProvider } from "../../../contexts/Auth/AuthProvider"
import { I18nextProvider } from "react-i18next"
import i18next from "i18next"

describe('Test in <About/>', () => {

    test('Component must render and match snapshot', () => {
        const { container } = render(
            <BrowserRouter>
                <AuthProvider>

                    <ThemeProvider theme={theme}>
                        <I18nextProvider i18n={i18next}>
                            <About />
                        </I18nextProvider>
                    </ThemeProvider>

                </AuthProvider>
            </BrowserRouter>
        )

        expect(container).toBeTruthy();
        expect(container).toMatchSnapshot();
    })

})