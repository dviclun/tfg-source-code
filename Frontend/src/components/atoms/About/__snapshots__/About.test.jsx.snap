// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <About/> Component must render and match snapshot 1`] = `<div />`;
